### PR TITLE
Support deleting table on V1 endpoint + FF fix

### PR DESCRIPTION
**Pull request by flvndvd**
## Description

This PR adds support to delete a structured table through the V1 endpoint. It also fixes the different places where we check for the feature flag.

### CODE REVIEW

File: front/pages/api/v1/w/[wId]/data_sources/[name]/tables/[tId]/index.ts [hunk 1]
```diff
@@ -37,7 +38,10 @@ async function handler(
     });
   }
 
-  if (!owner.flags.includes("structured_data")) {
+  if (
+    !owner.flags.includes("structured_data") ||
+    !owner.flags.includes("auto_pre_ingest_all_databases")
```
Comment (fontanierh) [inline front/pages/api/v1/w/[wId]/data_sources/[name]/tables/[tId]/index.ts:43]: Not sure we want this right ? this one should just condition whether we ingest all DBs automatically but we should still allow deleting regardless no ?

File: front/pages/api/w/[wId]/data_sources/[name]/tables/csv.ts [hunk 2]
```diff
@@ -100,7 +100,10 @@ export async function handlePostTableCsvUpsertRequest(
     });
   }
 
-  if (!owner.flags.includes("structured_data")) {
+  if (
+    !owner.flags.includes("structured_data") ||
+    !owner.flags.includes("auto_pre_ingest_all_databases")
```
Comment (fontanierh) [inline front/pages/api/w/[wId]/data_sources/[name]/tables/csv.ts:105]: same here I don't think we want that

File: front/pages/api/w/[wId]/data_sources/[name]/tables/index.ts [hunk 3]
```diff
@@ -40,7 +40,10 @@ async function handler(
     });
   }
 
-  if (!owner.flags.includes("structured_data")) {
+  if (
+    !owner.flags.includes("structured_data") ||
+    !owner.flags.includes("auto_pre_ingest_all_databases")
```
Comment (fontanierh) [inline front/pages/api/w/[wId]/data_sources/[name]/tables/index.ts:45]: and also same here

**Review by fontanierh — CHANGES_REQUESTED** [review]
LGTM except for the FF

**Comment by flvndvd** [comment]
Updated @fontanierh.

**Review by fontanierh — APPROVED** [review]
LGTM

**Pull request merged**